src/hooks/useReportIDs.tsx

```diff
@@ -0,0 +1,135 @@
+import React, {createContext, useCallback, useContext, useMemo} from 'react';
+import {withOnyx} from 'react-native-onyx';
+import type {OnyxCollection, OnyxEntry} from 'react-native-onyx';
+import {getCurrentUserAccountID} from '@libs/actions/Report';
+import {getPolicyMembersByIdWithoutCurrentUser} from '@libs/PolicyUtils';
+import SidebarUtils from '@libs/SidebarUtils';
+import CONST from '@src/CONST';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Beta, Policy, PolicyMembers, PriorityMode, Report, ReportActions, TransactionViolation} from '@src/types/onyx';
+import useActiveWorkspace from './useActiveWorkspace';
+import useCurrentReportID from './useCurrentReportID';
+
+type OnyxProps = {
+    chatReports: OnyxCollection<Report>;
+    betas: OnyxEntry<Beta[]>;
+    policies: OnyxCollection<Policy>;
+    allReportActions: OnyxCollection<ReportActions>;
+    transactionViolations: OnyxCollection<TransactionViolation[]>;
+    policyMembers: OnyxCollection<PolicyMembers>;
+    priorityMode: OnyxEntry<PriorityMode>;
+};
+
+type WithReportIDsContextProviderProps = OnyxProps & {
+    children: React.ReactNode;
+    currentReportIDForTests?: string;
+};
+
+type ReportIDsContextValue = {
+    orderedReportIDs: string[];
+};
+
+const ReportIDsContext = createContext<ReportIDsContextValue>({
+    orderedReportIDs: [],
+});
+
+function WithReportIDsContextProvider({
+    children,
+    chatReports,
+    betas,
+    policies,
+    allReportActions,
+    transactionViolations,
+    policyMembers,
+    priorityMode,
+    /**
+     * Only required to make unit tests work, since we
+     * explicitly pass the currentReportID in LHNTestUtils
+     * to SidebarLinksData, so this context doesn't have an
+     * access to currentReportID in that case.
+     *
+     * This is a workaround to have currentReportID available in testing environment.
+     */
+    currentReportIDForTests,
```

Comment:
I'm not a huge fan of modifying the code for the sake of testing. Is there a way to make the currentId context work in the test environment?

Reply:
by the way, could you link the unit test you're referring to? I'm not familiar with the architecture of unit testing in this repo.

Reply:
@hayata-suenaga  likewise but there is no possible way visible at the moment to achieve the same without adding this to the mock. This fails in `tests/perf-test/SidebarLinks.perf-test.js`. See https://github.com/Expensify/App/blob/c915e7f026eddc710dd01a38abc3afc9334002ae/tests/perf-test/SidebarLinks.perf-test.js#L109

This is a reassure test which fails because `currentReportID` is not available to it, so we have to manually supply the `currentReportID` which is valid only for testing.

Reply:
Thank you for the explanation 😄 

I dug into the test code a little bit (i.e. how the `MockedSiderLinks` component is defined) and found this line of code.

https://github.com/Expensify/App/blob/34a629d6b09f0de830c2f57319b0c813b899f831/tests/utils/LHNTestUtils.tsx#L283

I see the `CurrentReportIDContextProvider` provided to the mocked component. Doesn't this context provide a mocked current ID? 🤔

Reply:
No, this doesn't provide the mocked current ID. The reason is that in the code they are passing the currentID as a prop to `SidebarLinksData` and not through the context, hence we always have falsy value from the context.

https://github.com/Expensify/App/blob/34a629d6b09f0de830c2f57319b0c813b899f831/tests/utils/LHNTestUtils.tsx#L281-L295

And each time a Test want to change the reportID they call `getDefaultRenderedSidebarLinks(currentReportID)` which passes the `currentID` prop to `SidebarLinksData`

https://github.com/Expensify/App/blob/34a629d6b09f0de830c2f57319b0c813b899f831/tests/utils/LHNTestUtils.tsx#L300

Reply:
ah I can see that the test is taking the current report ID and then passing that to the prop.

in that case, do you happen to know why we wrap the component with the `CurrentReportIDContextProvider` for the test? 🤔 😄

Reply:
I don't know, probably to fix some edge cases?